File: src/app/models/permition.ts
export interface Estudiante {
  studentId: number;
  userId: number;
  nombreCompleto: string;
  email: string;
  telefono: string;
  numeroMatricula: string | null;
}

export interface Tutor {
  tutorId: number;
  userId: number;
  nombreCompleto: string;
  email: string;
  telefono: string;
}

export interface Profesor {
  teacherId: number;
  userId: number;
  nombreCompleto: string;
  email: string;
  telefono: string;
}

export interface Permition {
  permitId: number;
  estudiante: Estudiante;
  tutor: Tutor;
  profesores: Profesor[];
  startDate: string;
  endDate: string;
  reason: string;
  description: string;
  cuatrimestre: number;
  evidence: string;
  status: string;
  requestDate: string;
}

export interface PermitionsResponse {
  permits: Permition[];
  total: number;
}

File: src/main/kotlin/permition/domain/entities/Permition.kt
package permition.domain.entities

import java.time.LocalDate
import java.time.LocalDateTime

data class Permition(
    val permitId: Int? = null,
    val studentId: Int,
    val tutorId: Int,
    val teacherIds: List<Int> = emptyList(),
    val startDate: LocalDate,
    val endDate: LocalDate,
    val reason: PermitReason,
    val description: String,
    val cuatrimestre: Int,
    val evidence: String? = null,
    val status: PermitStatus = PermitStatus.PENDING,
    val requestDate: LocalDateTime = LocalDateTime.now(),
    val permitDocumentUrl: String? = null
)
